### ISSUE TITLE
Support for EntityFramework persistence

### ISSUE THREAD

**Issue by izzycoding** [mizrael/OpenSleigh#20]
Would it be useful to have an EntityFramework persistence package?
I might be able to look into developing one on my next project if it would be useful.
That would then enable various persistence options out of the box.

**Comment by mizrael** [comment]
I'll add it to my TODO list! I was thinking of tackling CosmosDB first, but I can switch.

**Comment by mizrael** [comment]
@izzycoding what do you mean by EntityFramework, specifically? Would a "standard" SQL persistence library work for you? OpenSleigh is abstracting away all the implementation details, do you specifically need it to use EF instead?

**Comment by izzycoding** [comment]
Ideally my use would need EntityFramework as our build system is already geared for building and deploying migrations. Also, we use MSSQL, SQLite, PostgreSQL and MySQL as potential database providers and we already have a framework for selecting and configuring the relevant EF provider at runtime.

**Comment by izzycoding** [comment]
For an EntityFramework implementation I would expect part of the service registration builder to allow the relevant EF serviceBuilder extensions to inject the provider. The DbContext is then an EF one which is registered similar to:
`cfg.UseEntityFrameworkPersistence(options => options.UseSqlServer(connString));`
Doing this would enable my code and CI/CD to work as it already does with minimal changes.

**Comment by izzycoding** [comment]
Also, noticed that EF has some providers being developed for MongoDB:
https://github.com/BlueshiftSoftware/EntityFrameworkCore
And CosmosDB:
https://docs.microsoft.com/en-us/ef/core/providers/cosmos/?tabs=dotnet-core-cli

A single implementation to cover 90% of use cases.

**Comment by mizrael** [comment]
well, as I wrote before, OpenSleigh would abstract all the implementation details, so it doesn't really matter what library it is using to access the persistence layer. It could be EF, Dapper, raw ADO, whatever.
From your perspective, you just have to feed the connection string when configuring the library in your Composition Root.

That said, EF is a valuable tool and would allow several different database technologies to be supported in (potentially) one go. I started already a PoC in [a feature branch](https://github.com/mizrael/OpenSleigh/tree/sql), I'll link this issue with the PR when I'm done.
